includes/Core/Authentication/Authentication.php

```diff
@@ -1401,14 +1415,34 @@ private function filter_features_via_proxy( $feature_enabled, $feature_name ) {
 	 *
 	 * @since 1.70.0
 	 *
+	 * @param string $feature_name Optional feature name that is specifically being checked.
 	 * @return array Array of features or an empty array if the fetch errored.
 	 */
-	private function get_transient_features() {
+	private function get_transient_features( $feature_name = '' ) {
 		$transient_name               = 'googlesitekit_remote_features';
 		$service_setup_v2_option_name = 'googlesitekitpersistent_service_setup_v2_enabled';
 
 		$features = $this->transients->get( $transient_name );
 		if ( false === $features ) {
+
+			if ( ! ! $feature_name ) {
+				$active_modules = $this->options->get( Modules::OPTION_ACTIVE_MODULES );
+				if ( ! is_array( $active_modules ) ) {
+					$active_modules = $this->options->get( 'googlesitekit-active-modules' );
+				}
+				if ( ! is_array( $active_modules ) ) {
+					$features[ $feature_name ]['enabled'] = false;
+					return $features;
+				}
+				if ( 'ideaHubModule' === $feature_name ) {
+					$features['ideaHubModule']['enabled'] = in_array( Idea_Hub::MODULE_SLUG, $active_modules, true );
+				}
+				if ( 'swgModule' === $feature_name ) {
+					$features['swgModule']['enabled'] = in_array( Subscribe_With_Google::MODULE_SLUG, $active_modules, true );
+				}
+				return $features;
+			}
```

Comment:
I think handling this feature-specific behavior in this method is particularly hacky, that's why my original proposal was to do it in the `filter_features_via_proxy` method. We can do it in _only that_ method, without modifying this private method here, which makes the hack less problematic since at least it's only in one place.

I think it would be cleaner to add the same `$this->transients->get` call to `filter_features_via_proxy` in case of `ideaHubModule` and `swgModule`, That is not an expensive call, so it's fine especially since it's only for these two modules. From there we then already know if the transient is set or not, so we can put this entire block of logic in only `filter_features_via_proxy`, leaving `get_transient_features` clean (especially the additional parameter here is problematic, since that doesn't only make the method internals "hacky", but also the method signature).

Reply:
Sorry I missed making an additional call to $this->transients->get() within filter_features_via_proxy() in your previous comment.